assets/src/edit-story/app/history/historyProvider.js

```diff
@@ -36,18 +38,38 @@ function HistoryProvider({ children, size }) {
     undo,
     redo,
   } = useHistoryReducer(size);
+  const canUndo = offset < historyLength - 1;
+  const changesSinceLastSave = useRef(0);
+  const [hasChangedSinceLastSave, setHasChangedSinceLastSave] = useState(false);
+
+  usePreventWindowUnload(hasChangedSinceLastSave);
+
+  useEffect(() => {
+    if (
+      historyLength > 0 &&
+      changesSinceLastSave.current !== historyLength - 1
```

Comment:
Note that when history reaches its buffer size, it'll prune from the other end and just stay at the same length continuously. Does this still work with a full history buffer?

Reply:
The length is currently set to only 50, so it'll be reached fairly quickly.

Maybe we want another flag coming from the history reducer with a _"history version number"_ or something like that - a number that increments when ever history changes.

@dvoytenko, does that make sense to you?

Reply:
Indeed. I also figured this problem after your first comment. I just pushed a proposal (which is basically what you said in order to create a global history length counter). Also, thanks for the report, I did not see this capability in the history changes.

Reply:
Indeed, `historyLength` is definitely a buffer thing. We need `updateCounter` or `versionId` or something that keeps incrementing forever in the editor's instance.